specs/kiota.config.md

```diff
@@ -0,0 +1,634 @@
+
+When executing, a new API entry will be added and will use the `--api-name` parameter as the key for the map. When loading the OpenAPI description, it will generate a hash of the description to enable change detection of the description and save it as part of the `descriptionHash` property. It will also store the location of the description in the `descriptionLocation` property. If `--include-path` or `--exclude-path` are provided, they will be stored in the `includePatterns` and `excludePatterns` properties respectively.
+
+| Parameters | Required | Example | Description |
+| -- | -- | -- | -- |
+| `--api-name \| --api` | Yes | graph | Name of the API |
+| `--openapi \| -d` | Yes | https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/openapi.yaml | The location of the OpenAPI description in JSON or YAML format to use to generate the SDK. Accepts a URL or a local path. |
+| `--include-path \| -i` | No | /me/chats#GET | A glob pattern to include paths from generation. Accepts multiple values. Defaults to no value which includes everything. |
+| `--exclude-path \| -e` | No | \*\*/users/\*\* | A glob pattern to exclude paths from generation. Accepts multiple values. Defaults to no value which excludes nothing. |
+
+#### Using `kiota api add`
+
+```bash
+kiota api add --api-name "graph" --openapi "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/openapi.yaml" --include-path "/me/chats#GET" --include-path "/me#GET"
+```
+
+```javascript
+// Adds the following to the kiota.config file
+"graph": {
+  "descriptionHash": "9EDF8506CB74FE44...",
+  "descriptionLocation": "https://.../openapi.yaml",
+  "includePatterns": ["/me/chats#GET", "/me#GET"],
+  "excludePatterns": []
+}
+```
+
+The resulting `kiota.config` file will look like this:
+
+```javascript
+{
+  "name": "My application",
+  "apis": {
+    "graph": {
+      "descriptionHash": "9EDF8506CB74FE44...",
+      "descriptionLocation": "https://.../openapi.yaml",
+      "includePatterns": ["/me/chats#GET", "/me#GET"],
+      "excludePatterns": []
+    }
+  }
+}
+```
+
+### kiota api delete
+
+`kiota api delete` allows a developer to delete an existing API from the `kiota.config` file. The command will remove the entry from the `apis` section of the `kiota.config` file. The command has one required parameter, the name of the API (key of the apis map). The command also has one optional parameter, the ability to remove generated clients. If provided, kiota will delete the folder specified at the `outputPath` from the client configuration.
+
+| Parameters | Required | Example | Description |
+| -- | -- | -- | -- |
+| `--api-name \| --api` | Yes | graph | Name of the API |
+| `--clean-output \| --co` | No |  | Cleans the generated clients from the API |
+
+#### Using kiota api delete
```

Comment:
removing a client/api should also remove the source if the API came from a registry